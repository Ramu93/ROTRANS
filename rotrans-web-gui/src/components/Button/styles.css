.txt-btn-div {
  height: 40px;
  width: 160px;
  background-color: #6435c9;
  display: flex;
  align-items: center;
  justify-content: center;
  border-radius: 4px;
  border-top-right-radius: 0px;
  border-bottom-right-radius: 0px;
}

.icon-btn-div {
  height: 40px;
  width: 40px;
  display: flex;
  align-items: center;
  justify-content: center;
}

.custom-btn-div {
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;
}

.custom-btn {
  margin-top: 32px;
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;
  border: 0;
  background-color: #6435c9;
  border-radius: 4px;
  cursor: pointer;
}

.btn-label {
  color: white;
  font-size: 14px;
}

.btn-icon {
  padding-top: 4px;
}
