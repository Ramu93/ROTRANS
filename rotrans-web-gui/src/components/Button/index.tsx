import React from "react";
import "./styles.css";

interface Props {
  label?: string;
  icon?: any;
  disabled?: boolean;
  onClick?:
    | ((event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void)
    | undefined;
  iconButton?: boolean;
}

const Button: React.FC<Props> = ({ label, icon, onClick, disabled, iconButton }) => {
  return (
    <div className="custom-btn-div">
      <div className="custom-btn-inner-div">
        <button className="custom-btn" onClick={onClick} disabled={disabled}>
          {label && (
            <div className="txt-btn-div">
              <span data-testid="label" className="btn-label">
                {" "}
                {label}
              </span>
            </div>
          )}
          {icon && (
            <div className="icon-btn-div">
              <span data-testid="icon" className="btn-icon">
                {icon}
              </span>
            </div>
          )}
        </button>
      </div>
    </div>
  );
};

export default Button;
