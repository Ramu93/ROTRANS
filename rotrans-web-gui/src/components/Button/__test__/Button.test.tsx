import React from "react";
import ReactDOM from "react-dom";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom";
import renderer from "react-test-renderer";
import Button from "..";

describe("Button component", () => {
  afterEach(cleanup);

  it("Rendered", () => {
    const div = document.createElement("div");
    ReactDOM.render(<Button label="Transfer" />, div);
  });

  it("Rendered with props", () => {
    const { getByTestId } = render(<Button label="Transfer" />);
    expect(getByTestId("label")).toHaveTextContent("Transfer");
  });

  it("Matches snapshot 1 - without icon", () => {
    const tree = renderer.create(<Button label="Transer" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Matches snapshot 2 - with icon", () => {
    const tree = renderer
      .create(
        <Button
          label="Transfer"
          icon={<img src={require("../../../assets/svg/go.svg")} />}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
