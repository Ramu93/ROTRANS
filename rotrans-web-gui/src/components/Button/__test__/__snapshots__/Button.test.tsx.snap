// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component Matches snapshot 1 - without icon 1`] = `
<div
  className="custom-btn-div"
>
  <div
    className="custom-btn-inner-div"
  >
    <button
      className="custom-btn"
    >
      <div
        className="txt-btn-div"
      >
        <span
          className="btn-label"
          data-testid="label"
        >
           
          Transer
        </span>
      </div>
    </button>
  </div>
</div>
`;

exports[`Button component Matches snapshot 2 - with icon 1`] = `
<div
  className="custom-btn-div"
>
  <div
    className="custom-btn-inner-div"
  >
    <button
      className="custom-btn"
    >
      <div
        className="txt-btn-div"
      >
        <span
          className="btn-label"
          data-testid="label"
        >
           
          Transfer
        </span>
      </div>
      <div
        className="icon-btn-div"
      >
        <span
          className="btn-icon"
          data-testid="icon"
        >
          <img
            src={
              Object {
                "ReactComponent": Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "render": [Function],
                },
                "__esModule": true,
                "default": "go.svg",
              }
            }
          />
        </span>
      </div>
    </button>
  </div>
</div>
`;
